evm_arithmetization/src/fixed_recursive_verifier.rs

```diff
@@ -2944,3 +2944,101 @@ where
     circuit.verifier_only.circuit_digest.elements.len()
         + (1 << circuit.common.config.fri_config.cap_height) * NUM_HASH_OUT_ELTS
 }
+
+#[cfg(test)]
+mod tests {
+    use plonky2::field::goldilocks_field::GoldilocksField;
+    use plonky2::plonk::config::PoseidonGoldilocksConfig;
+    use plonky2::timed;
+
+    use super::*;
+    use crate::testing_utils::{dummy_payload, init_logger};
+
+    type F = GoldilocksField;
+    const D: usize = 2;
+    type C = PoseidonGoldilocksConfig;
+
+    #[test]
+    #[ignore]
+    fn test_segment_proof_generation_without_keccak() -> anyhow::Result<()> {
+        let timing = &mut TimingTree::new("Segment Proof Generation", log::Level::Info);
+        init_logger();
+
+        let all_stark = AllStark::<F, D>::default();
+        let config = StarkConfig::standard_fast_config();
+
+        let all_circuits = timed!(
+            timing,
+            log::Level::Info,
+            "Create all recursive circuits",
+            AllRecursiveCircuits::<F, C, D>::new(
+                &all_stark,
+                &[
+                    16..17,
+                    8..9,
+                    9..10,
+                    4..5,
+                    8..9,
+                    4..5,
+                    17..18,
+                    17..18,
+                    17..18
+                ],
+                &config,
+            )
+        );
+
+        // Generate a dummy payload for testing
+        let dummy_payload = timed!(
+            timing,
+            log::Level::Info,
+            "Generate dummy payload",
+            dummy_payload(100, true)?
+        );
+
+        let max_cpu_len_log = 9;
+        let segment_iterator = SegmentDataIterator::<F>::new(&dummy_payload, Some(max_cpu_len_log));
+
+        let mut proofs_without_keccak = vec![];
+
+        let skip_proofs_before_index = 3;
+        for (i, segment_run) in segment_iterator.enumerate() {
+            if i < skip_proofs_before_index {
+                continue;
+            }
```

Comment:
This seems highly specific, as any modification in the KERNEL ASM may break your assumption that the 3rd segment has no keccak operation.
Also this is breaking compilation with `cdk_erigon` feature flag, as in this case we're missing a range for the Poseidon table in the initialization of `AllRecursiveCircuits`

Reply:
Yes, I agree. Do you have any suggestions on how to craft a payload that generates segments without Keccak? Regarding cdk_erigon, I can add a feature flag to disable it.

Reply:
Hmm, I think we'd need some around segment generation. Basically we'd want to do the same as in `init_exc_stop.rs` but I don't think this will play well with how we actually generate segment data.